packages/babel-plugin-import-jsx-pragma/src/index.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Default options for the plugin.
+ *
+ * @property {string}  scopeVariable Name of variable required to be in scope
+ *                                   for use by the JSX pragma. For the default
+ *                                   pragma of React.createElement, the React
+ *                                   variable must be within scope.
+ * @property {string}  source        The module from which the scope variable
+ *                                   is to be imported when missing.
+ * @property {boolean} isDefault     Whether the scopeVariable is the default
+ *                                   import of the source module.
+ */
+const DEFAULT_OPTIONS = {
+	scopeVariable: 'React',
+	source: 'react',
+	isDefault: true,
+};
+
+/**
+ * Babel transform plugin for automatically injecting an import to be used as
+ * the pragma for the React JSX Transform plugin.
+ *
+ * @see http://babeljs.io/docs/en/babel-plugin-transform-react-jsx
+ *
+ * @param {Object} babel Babel instance.
+ *
+ * @return {Object} Babel transform plugin.
+ */
+export default function( babel ) {
+	const { types: t } = babel;
+
+	let _options;
+
+	let hasJSX, hasImportedScopeVariable;
+
+	function getOptions( options ) {
+		if ( ! _options ) {
+			_options = {
+				...DEFAULT_OPTIONS,
+				...options,
+			};
+		}
+
+		return _options;
+	}
+
+	return {
+		visitor: {
+			JSXElement() {
+				hasJSX = true;
```

Comment:
I read somewhere in babel docs that it's not great to rely on state variables like that and that it's probably better to rely on the "state" object passed to the visitors because of the way these visitors can be called in random orders.

I'm not a babel-plugin developer so I can't argue if it's legit here or not, just wanted to mention it.

Reply:
Based on the handbook description of visiting behavior, and particularly with its evaluation on the `Program.exit` pass, I don't know that it should be a concern.

https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#visitors

That said, I'll take a look to see if it's easy enough to switch to using state.

Reply:
Updated in 67653d0. Makes things simpler overall too.